### PR TITLE
CLI: Add wasmatic commands

### PR DESCRIPTION
**Pull request by ueco-jb**

### CODE REVIEW

File: tools/cli/config.json
Comment (ueco-jb) [inline tools/cli/config.json:9]: It will be gone after rebase, and rebase will be gone after merge conflits are solved...

File: wasi/square/src/bindings.rs
Comment (ueco-jb) [inline wasi/square/src/bindings.rs:1]: Hey, don't blame me.

**Comment by ueco-jb** [comment]
The code is simple right now and arguments are not validated yet. 
I had a lot of issues figthing with the respones and json parsing for some reson...
You can see that the deploy works in both cases:

```bash
$ avs-toolkit-cli wasmatic deploy \
--name test-btc \
--digest 'sha256:486e42639144dbd48364fb0ec68846bfc0b45de5777e5e377f5496d91b9abec3' \
--wasm-source 'https://raw.githubusercontent.com/macovedj/test/main/btc-avg/btc_avg.wasm' \
--trigger '{ "cron": { "schedule": "1/15 * * * * *" } }' \
--envs '[["API_KEY", "x-cg-demo-api-key=CG-PsTvxDqXZP3RD4TWNxPFamcW"]]'`
```
and from the file
```bash
wasmatic deploy \
  --name square-example \
  --digest 'sha256:d902444dcc241c6d19fefbdaf2e34d3d88d17453f11450ae605c4e9357c5bd14' \
  --wasm-source ./cavs_square.wasm \
  --trigger '{ "cron": { "schedule": "1/15 * * * * *" } }' \
  --envs '[["API_KEY", "x-cg-demo-api-key=CG-PsTvxDqXZP3RD4TWNxPFamcW"]]'
```
For some reason I'm getting the error 
`Error: Error: "{\"message\":\"ParseSchedule\"}"` but the message goes through and the binary is deployed.

Then, when I try to remove it, I'm getting another error:
```
Error: Error: "{\"message\":\"app not active\"}"
```
I need to investigate it further, but feel free to review @ethanfrey

**Comment by ethanfrey** [comment]
I will take a look. I prefer having simpler args and not just the copy of JSON.
Here are some concrete requests for the arg parsing (I will review functionality now)

1. Make digest optional for local file and just calculate it in the CLI
2. envs should be more like normal arg format, and a repeated flag `--envs "API_KEY=x-cg-demo-api-key=CG-PsTvxDqXZP3RD4TWNxPFamcW" --envs MY_SECRET=foobar`
3. For setting up cron, how about `--cron-schedule = "1/15 * * * * *"` (but should it be 7 stars?)
4. For setting up task queue, how about `--task-queue=layer1foobarbaz`

File: wasi/square/src/bindings.rs
Comment (ethanfrey) [reply wasi/square/src/bindings.rs:1]: Oh, you gotta run `cargo fmt` every time after `cargo component build`...
I put that into the `build_wasi.sh` script, but it might need a rebase to get that version

**Review by ethanfrey — APPROVED** [review]
I have not tested, but the code looks good.
Can you please simplify the cli args, and do the rebase fixups.?
After that good enough to merge.
Though I might want to revisit the cron format and test it a bit

**Comment by ueco-jb** [comment]
Ready to merge, the comments will be addressed shortly in a follow up.

File: tools/cli/src/commands/wasmatic.rs
Comment (ueco-jb) [reply tools/cli/src/commands/wasmatic.rs:132]: Yeah, I noticed even if it fails it show the "success" result because it receives 200 http resposne with the error inside.

File: tools/cli/src/args.rs
Comment (ethanfrey) [reply tools/cli/src/args.rs]: fix in 161002819

File: tools/cli/src/args.rs
Comment (ethanfrey) [reply tools/cli/src/args.rs]: fixed in https://github.com/Lay3rLabs/avs-toolkit/commit/1610028194a50ac08a298276326f34e2a988af04

File: tools/cli/src/commands/wasmatic.rs
Comment (ethanfrey) [reply tools/cli/src/commands/wasmatic.rs:158]: Fixed in https://github.com/Lay3rLabs/avs-toolkit/pull/47/commits/2088d44e22229415159e68f9a6b2eff9030e2cca

Along with improving the arguments

**Review by ethanfrey — APPROVED** [review]
Approved with my fixes.

Feel free to merge this once you ran through my demo

**Comment by ueco-jb** [comment]
Works for me, except I'm getting that 404 on /test. Probably I messed something with the images? 
Merging now, I will expand the cli plus address any following issues with the next changeset.
Easier to ask people to just pull the latest main.

**Pull request merged**
